### PR TITLE
Extend expanse model with subcategory

### PR DESCRIPTION
**Pull request by dkwasniewski**

### CODE REVIEW

File: src/configs/categories.config.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 const TYPES = {
     EARNINGS: 'EARNINGS',
-    EXPENSES: 'EXPENSES'
+    EXPENSES: 'EXPENSES',
+    SAVINGS: 'SAVINGS'
```
Comment (piontek555) [inline src/configs/categories.config.js]: Missing comma on end of line

Comment (dkwasniewski) [reply src/configs/categories.config.js]: added

File: src/routers/expenses.js [hunk 2]
```diff
@@ -1,10 +1,35 @@
...
+    CategoryWithSubCategories.findOne(
+        { name: categoryName },
+        ['subCategories', 'type'],
+        function (err, docs) {
+            if (err) { return res.status(400).send(err) }
+            const isSubCategory = docs.subCategories.some(sub => sub === subCategory)
+            if(!isSubCategory) {
+                return res.status(400).send(`SubCategory ${subCategory} was not found for category ${categoryName}`)
+            }
+        },
```
Comment (GrzegorzPerko) [inline src/routers/expenses.js:31]: Raz jest przecinek raz nima

Comment (dkwasniewski) [reply src/routers/expenses.js:31]: poprawione, dodałem eslinta i pre-commit hooki

File: .eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+module.exports = {
+    'env': {
+        'node': true,
+        'commonjs': true,
+        'es6': true
```
Comment (konradperko) [inline .eslintrc.js:5]: can't it be es7, or8? just wondering

File: .eslintrc.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        'ecmaVersion': 2018
+    },
+    'rules': {
+        "comma-dangle": [
+            "error",
+            "always-multiline"
+        ],
+        'indent': [
+            'error',
+            4
```
Comment (konradperko) [inline .eslintrc.js:22]: 2?

**Review by cezaryintive — APPROVED** [review]

**Review by konradperko — APPROVED** [review]

File: .eslintrc.js (hunk 4 above)
Comment (dkwasniewski) [reply .eslintrc.js:22]: you are in the charge, 4 is fine, I'm just trolling :P it is all about habits

File: .eslintrc.js (hunk 3 above)
Comment (dkwasniewski) [reply .eslintrc.js:5]: I am not 100% sure. It was one of the options during configuration. Here are some other EcmaScript version possibilities: https://eslint.org/docs/user-guide/configuring#specifying-parser-options

**Pull request merged**